### PR TITLE
add: getting started on solana intro

### PR DESCRIPTION
**Pull request by jacobcreech**

### CODE REVIEW

File: docs/intro/dev.md [hunk 1]
```diff
@@ -1,238 +1,178 @@
...
+The "glue" between working on the client-side and the on-chain side is the
+[JSON RPC API](https://solana.com/docs/rpc). Developers working on the
+client-side send RPC requests to the Solana network to interact with on-chain
+programs. This is very similar to normal development between a frontend and
+backend. The major difference with working on Solana is that the backend is a
+global permissionless blockchain. This means that anyone can interact with your
+on-chain program without the need of issuing API keys or any other form of
+permission.
+
+![A view of how clients work with the Solana blockchain](/assets/guides/intro/developer_flow.png)
```
Comment (jacobcreech) [inline docs/intro/dev.md:41]: Not a blocker imo but I'm asking Hovin to make a nice picture of this

**Review by nickfrosty — CHANGES_REQUESTED** [review]
We can go back and update the graphic later after Hovin creates it too

File: docs/intro/dev.md [hunk 2]
```diff
@@ -1,238 +1,178 @@
...
+accelerates your development and makes developing easy. You can read more about
+[Anchor](https://www.anchor-lang.com/) on their website.
+
+You'll need a way to test your program. There are a few different ways to test
+your program based on your language preference:
+
+- [solana-program-test](https://docs.rs/solana-program-test/latest/solana_program_test/) -
+  Testing framework built in Rust
+- [solana-bankrun](https://kevinheavey.github.io/solana-bankrun/) - Testing
+  framework built in Python or Typescript
```
[the remark this answers is not in the document]
Comment (jacobcreech) [reply docs/intro/dev.md:139]: Ah, it got moved under solders with the same module name https://github.com/kevinheavey/solders

File: docs/intro/dev.md [hunk 3]
```diff
@@ -1,238 +1,180 @@
...
+  you through building on Solana.
 
-Devnet has a faucet that allows you to get free SOL to test with. It costs $0 to
-do development on Solana.
+## Getting Support
 
-Check out the [clusters page](/docs/core/clusters.md) for more information on
-these.
+The best support you can find is by asking your question on the
+[Solana StackExchange](https://solana.stackexchange.com/).
```
Comment (mikemaccana) [inline docs/intro/dev.md:178]:
```suggestion
The best support you can find is on [Solana StackExchange](https://solana.stackexchange.com/). Search for your question there first - there's a good chance there will already be a question asked by someone else, with an answer. If it's not there, add a new question! Remember to include as much detail as you can in your question, and please use text (not screenshots) to show error messages, so other people with the same problem can find your question!

```

Comment (mikemaccana) [reply docs/intro/dev.md:178]: Just a suggestion. Maybe this is too much detail. WDYT?

File: public/assets/docs/intro/developer_flow.png
Comment (mikemaccana) [inline public/assets/docs/intro/developer_flow.png:1]: I'd re-do this in Whimsical, should only take a moment.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Excellent, this includes so many things I wish I had known when getting started on Solana, like the local validator and how similar Solana development is to traditional backend/frontend work. 

Most of these fixes are just minor English bits and some terminology consistency fixes. Also I'm not totally sure on how well C / C++ is maintained compared to Rust - if it's just as good, great, if not I think we should just point people to Rust. Changes are done as suggestions so you can just click the button if you agree.

File: docs/intro/dev.md [hunk 4]
```diff
@@ -1,238 +1,180 @@
...
+development works, and provide the critical tools you'll need to get started.
+
+## High Level Developer Overview
+
+Development on Solana can be broken down into two main parts:
+
+1. **On-chain Program Development**: This is where you create and deploy custom
+   programs directly to the blockchain. Once deployed, anyone who knows how to
+   communicate with them can use them. You can write these programs in Rust, C,
+   or C++.
```
[the remark this answers is not in the document]
Comment (jacobcreech) [reply docs/intro/dev.md:27]: Worth mentioning, but we should have a recommendation to use Rust as the best maintained.

File: public/assets/docs/intro/developer_flow.png
Comment (jacobcreech) [reply public/assets/docs/intro/developer_flow.png:1]: Getting an updated version from Hovin. Solana-themed

**Comment by jacobcreech** [comment]
All comments should be addressed now

**Review by nickfrosty — APPROVED** [review]
I fixed a few last minute types from the recent changes, but the rest is great!

**Pull request merged**